php-checks/src/main/java/org/sonar/php/checks/regex/AbstractRegexCheck.java

```diff
@@ -34,15 +36,20 @@
 import org.sonar.plugins.php.api.tree.expression.FunctionCallTree;
 import org.sonar.plugins.php.api.tree.expression.LiteralTree;
 import org.sonar.plugins.php.api.tree.expression.VariableIdentifierTree;
+import org.sonar.plugins.php.api.tree.lexical.SyntaxToken;
 import org.sonar.plugins.php.api.visitors.CheckContext;
 import org.sonar.plugins.php.api.visitors.PhpIssue;
 import org.sonar.plugins.php.api.visitors.PreciseIssue;
 import org.sonarsource.analyzer.commons.regex.RegexParseResult;
 import org.sonarsource.analyzer.commons.regex.ast.FlagSet;
 import org.sonarsource.analyzer.commons.regex.ast.RegexSyntaxElement;
 
+import static org.sonar.php.regex.PhpRegexSource.BRACKET_DELIMITERS;
+
 public abstract class AbstractRegexCheck extends FunctionUsageCheck implements RegexCheck {
 
+  private static final Pattern DELIMITER_PATTERN = Pattern.compile("^[^\\w\\r\\n\\t\\f\\v ]");
```

Comment:
What about the following line in the [PHP reference](https://www.php.net/manual/en/regexp.reference.delimiters.php)?

> Leading whitespace before a valid delimiter is silently ignored

We could handle it as a separate ticket.

Reply:
Good point. I was aware of this case but forget to create a ticket for this. Thanks.